### PR TITLE
{Do not merge} open for edge build

### PR DESCRIPTION
**Pull request by MyronFanQiu**
**Description**  

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
edge build

File: src/azure-cli/azure/cli/command_modules/monitor/commands.py [hunk 1]
```diff
@@ -277,15 +347,39 @@ def load_command_table(self, _):
...
+
+    with self.command_group('monitor log-analytics cluster', log_analytics_cluster_sdk, custom_command_type=log_analytics_cluster_custom, is_preview=True) as g:
+        g.custom_command('create', 'create_log_analytics_cluster', validator=make_cluster_creation_as_no_wait_by_default)
+        g.custom_command('update', 'update_log_analytics_cluster')
+        g.show_command('show', 'get')
+        g.command('delete', 'delete', confirmation=True, supports_no_wait=True)
+        g.custom_command('list', 'list_log_analytics_clusters')
+        g.wait_command('wait')
+
+    with self.command_group('monitor log-analytics workspace linked-storage', log_analytics_linked_storage_sdk, custom_command_type=log_analytics_linked_storage_custom, is_preview=True) as g:
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/commands.py:376]: As "with self.command_group('monitor', metric_alert_sdk, custom_command_type=monitor_general_custom, is_preview=True) " already defined as `preview`, can we just remove "is_preview=True" here?

File: src/azure-cli/azure/cli/command_modules/monitor/commands.py [hunk 2]
```diff
@@ -277,15 +347,39 @@ def load_command_table(self, _):
...
+        g.command('create', 'create_or_update')
+        g.custom_command('add', 'add_log_analytics_workspace_linked_storage_accounts')
+        g.custom_command('remove', 'remove_log_analytics_workspace_linked_storage_accounts')
+        g.command('delete', 'delete')
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/commands.py:380]: add confirm

File: src/azure-cli/azure/cli/command_modules/monitor/operations/log_analytics_workspace.py [hunk 3]
```diff
@@ -2,29 +2,37 @@
...
 # --------------------------------------------------------------------------------------------
-from azure.mgmt.loganalytics.models import SkuNameEnum
+from azure.mgmt.loganalytics.models import WorkspaceSkuNameEnum
 
 
 def create_log_analytics_workspace(cmd, client, resource_group_name, workspace_name, location=None, tags=None,
-                                   sku=SkuNameEnum.per_gb2018.value, retention_time=None):
-    from azure.mgmt.loganalytics.models import Workspace, Sku
+                                   sku=WorkspaceSkuNameEnum.per_gb2018.value, retention_time=None,
+                                   public_network_access_for_query=None, public_network_access_for_ingestion=None):
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/operations/log_analytics_workspace.py:10]: public_network_access_for_query is None by default?

File: src/azure-cli/azure/cli/command_modules/monitor/_params.py [hunk 4]
```diff
@@ -339,12 +339,53 @@ def load_arguments(self, _):
...
+        c.argument('key_name', help='The name of the key associated with the Log Analytics cluster.')
+        c.argument('key_version', help='The version of the key associated with the Log Analytics cluster.')
+    # endregion
+
+    # region Log Analytics Linked Storage Account
+    with self.argument_context('monitor log-analytics workspace linked-storage') as c:
+        from azure.mgmt.loganalytics.models import DataSourceType
+        c.argument('data_source_type', help='Data source type for the linked storage account.',
+                   options_list=['--type'], arg_type=get_enum_type(DataSourceType))
+        c.argument('storage_account_ids', nargs='+', options_list=['--storage-accounts'],
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_params.py:384]: will it be required for create_or_update?

File: src/azure-cli/azure/cli/command_modules/monitor/operations/log_analytics_linked_storage_account.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+                                   storage_account_ids=linked_storage_accounts.storage_account_ids)
+
+
+def remove_log_analytics_workspace_linked_storage_accounts(client, resource_group_name, workspace_name,
+                                                           data_source_type, storage_account_ids):
+    linked_storage_accounts = client.get(resource_group_name=resource_group_name,
+                                         workspace_name=workspace_name,
+                                         data_source_type=data_source_type)
+    for storage_account_id in storage_account_ids:
+        if storage_account_id in linked_storage_accounts.storage_account_ids:
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/operations/log_analytics_linked_storage_account.py:25]: >in [](start = 30, length = 2)

is it case-insensitive?

File: src/azure-cli/azure/cli/command_modules/monitor/_help.py [hunk 6]
```diff
@@ -985,6 +1043,120 @@
...
+examples:
+  - name: Delete all linked storages with a specific type for a log analytics workspace
+    text: az monitor log-analytics workspace linked-storage delete --type AzureWatson -g MyResourceGroup --workspace-name MyWorkspace
+"""
+
+helps['monitor log-analytics workspace linked-storage add'] = """
+type: command
+short-summary: Add linked storages for log analytics workspace.
+examples:
+  - name: Add a linked storage for a log analytics workspace
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_help.py:1131]: not `a`, but `two'

File: src/azure-cli/azure/cli/command_modules/monitor/_help.py [hunk 7]
```diff
@@ -985,6 +1043,120 @@
...
+type: command
+short-summary: Delete all linked storages with specific data source type for log analytics workspace.
+examples:
+  - name: Delete all linked storages with a specific type for a log analytics workspace
+    text: az monitor log-analytics workspace linked-storage delete --type AzureWatson -g MyResourceGroup --workspace-name MyWorkspace
+"""
+
+helps['monitor log-analytics workspace linked-storage add'] = """
+type: command
+short-summary: Add linked storages for log analytics workspace.
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_help.py:1129]: with specific data source type

File: src/azure-cli/azure/cli/command_modules/monitor/_help.py [hunk 8]
```diff
@@ -985,6 +1043,120 @@
...
+examples:
+  - name: Add a linked storage for a log analytics workspace
+    text: az monitor log-analytics workspace linked-storage add --type AzureWatson -g MyResourceGroup --workspace-name MyWorkspace --storage-accounts SA1 SA2
+"""
+
+helps['monitor log-analytics workspace linked-storage remove'] = """
+type: command
+short-summary: Remove all linked storages with specific data source type for log analytics workspace
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_help.py:1137]: not `all`, but `some`

File: src/azure-cli/azure/cli/command_modules/monitor/_help.py [hunk 9]
```diff
@@ -985,6 +1043,120 @@
...
+examples:
+  - name: Remove a linked storage for a log analytics workspace
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_help.py:1139]: `a` to `two`

File: src/azure-cli/azure/cli/command_modules/monitor/_help.py [hunk 10]
```diff
@@ -985,6 +1043,120 @@
...
+type: command
+short-summary: Place the CLI in a waiting state until a condition of the linked service is met.
+examples:
+  - name: Pause executing next line of CLI script until the linked service is successfully provisioned.
+    text: az monitor log-analytics workspace linked-service wait -n cluster -g MyResourceGroup --workspace-name MyWorkspace --created
+"""
+
+helps['monitor log-analytics workspace linked-storage'] = """
+type: group
+short-summary: Manage linked storage for log analytics workspace.
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_help.py:1108]: linked storage account is more accurate?

File: src/azure-cli/azure/cli/command_modules/monitor/_params.py (hunk 4 above)
Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/monitor/_params.py:384]: required. I think it make senses. Creating a linked storage without storage account is quite strange.

**Pull request closed without merging**